### PR TITLE
License: Mark instructions GPLv3 and everything else MIT

### PR DESCRIPTION
**Pull request by ckamm**
To do that, split up the Accounts objects and the instruction implementations.

GPL code is only used when the "enable-gpl" feature is enabled. That means compiling the program or running tests need explicit feature activation now.

### CODE REVIEW

File: programs/mango-v4/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![allow(unused_variables)]
```
Comment (microwavedcola1) [inline programs/mango-v4/src/lib.rs:1]: curious, why was this needed?

**Review by microwavedcola1 — APPROVED** [review]

File: programs/mango-v4/src/lib.rs (hunk 1 above)
Comment (ckamm) [reply programs/mango-v4/src/lib.rs:1]: When compiled without `enable-gpl`, none of the instruction function arguments will end up being used - unfortunately I didn't find a way to conditionally enable this flag.
